http-webhook.md

```diff
@@ -0,0 +1,339 @@
+not support the feature, the sender MAY choose not to send to the target, at
+all, or send only at a very low request rate.
+
+### 4.1.1. Validation request
+
+The validation request uses the HTTP [OPTIONS][OPTIONS] method. The request
+is directed to the exact resource target URI that is being registered.
+
+With the validation request, the sender asks the target for permission to
+send notifications, and it can declare a desired request rate (requests per
+minute).
+
+The delivery target will respond with a permission statement and the permitted
+request rate.
+
+The following header fields MUST be included in the validation request:
+
+#### 4.1.1.2. WebHook-Request-Origin
+
+The `WebHook-Request-Origin` header requests permission to send notifications
+from this sender, and contains a DNS expression that identifies the sending
+system, for example "eventemitter.example.com". The value is meant to summarily
+identify all sender instances that act on the behalf of a certain system, and
+not an individual host.
+
+After the handshake and when permission is granted, the sender MUST use the
+`Origin` request header for each delivery request, with the value matching that
+of this header.
+
+For example:
+
+``` text
+WebHook-Request-Origin: eventemitter.example.com
+```
+
+#### 4.1.1.2. WebHook-Request-Rate
+
+The `WebHook-Request-Rate` header MAY be included in the request and asks for
+permission to send notifications from this sender at the specified rate. The
+value is the string representation of a positive integer number greater than
+zero and expresses the request rate in "requests per minute".
+
+For example, the following header asks for permission to send 120 requests per
+minute:
+
+``` text
+WebHook-Request-Rate: 120
+```
+
+### 4.1.2 Validation response
+
+If and only if the delivery target does allow delivery of the events, it MUST
+reply to the request by including the `WebHook-Allowed-Origin` and
+`WebHook-Allowed-Rate` headers.
+
+If it does not allow delivery of the events or does not expect delivery of
+events and nevertheless handles the HTTP OPTIONS method, the existing response
+ought not to be interpreted as consent, and therefore the handshake cannot
+rely on status codes.
+
+The response SHOULD include the [Allow][Allow] header indicating the [POST][POST]
+method being permitted. Other methods MAY be permitted on the resource, but
+their function is outside the scope of this specification.
+
+#### 4.1.1.2. WebHook-Allowed-Origin
+
+The `WebHook-Allowed-Origin` header MUST be returned when the delivery target
+agrees to notification delivery by the origin service. Its value MUST either be
+the origin name supplied in the `WebHook-Request-Origin` header, or a singular
+asterisk character ('*'), indicating that the delivery target supports
+notifications from all origins.
+
+``` text
+WebHook-Allowed-Origin: eventemitter.example.com
+```
+
+or
+
+``` text
+WebHook-Request-Origin: *
+```
+
+#### 4.1.1.2. WebHook-Allowed-Rate
```

Comment:
I already call out usage of 429 in section 2.2. 

The problem I am trying to address here is that the publisher doesn't suddenly show up with a mortal flood of events that takes out the target outright. 429 is reactive. This here aims to prevent the negotiation apriori.